### PR TITLE
Add simulator tracing, and debug port configuration via environment v…

### PR DESCRIPTION
**Pull request by ronlieb**
Cleaned u stderr in fprintfs, fixed typos in closing )s.

### CODE REVIEW

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 1]
```diff
@@ -29,6 +29,39 @@ int init_sim() {
...
+    if (T && T[0] != 0) {
+        status - sim->SetTracing(HEX_TRACE_PC_MIN, T);
+        if (status != HEX_STAT_SUCCESS) {
+            printf("HexagonWrapper::SetTracing MIN failed: %d\n", status);
+            return -1;
+        }
+    } else {
+        const char *T = getenv("HALIDE_HEX_SIM_TRACE");
+        if (T && T[0] != 0) {
+             status - sim->SetTracing(HEX_TRACE_PC, T);
```
Comment (dsharlet) [inline src/runtime/hexagon_remote/sim_host.cpp]: Extra space indentation

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 2]
```diff
@@ -29,6 +29,39 @@ int init_sim() {
...
+
+    // Configure use of debugger
+    int pnum = 0;
+    char *s= getenv("HALIDE_HEX_DBG_PORT");
+    if (s && (pnum=(atoi(s)))) {
+        fprintf(stderr, "\nDebugger port: %d\n", pnum);
+        HEXAPI_Status dbg_state;
+        dbg_state = sim->ConfigureRemoteDebug(pnum);
+        if (dbg_state) {
+            printf("\n`Debugger port failed: state: %d\n", dbg_state);
```
Comment (dsharlet) [inline src/runtime/hexagon_remote/sim_host.cpp]: There's a weird character after the \n, can you remove the newline and the weird character?

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 3]
```diff
@@ -29,6 +29,39 @@ int init_sim() {
         return -1;
     }
 
+    // Configue tracing.
+    const char *T = getenv("HALIDE_HEX_SIM_MIN_TRACE");
+    if (T && T[0] != 0) {
+        status - sim->SetTracing(HEX_TRACE_PC_MIN, T);
```
Comment (dpalermo) [inline src/runtime/hexagon_remote/sim_host.cpp]: Should this be 'status = '

File: src/runtime/hexagon_remote/sim_host.cpp (hunk 1 above)
Comment (dpalermo) [inline src/runtime/hexagon_remote/sim_host.cpp]: Should this be 'status = '

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 4]
```diff
@@ -29,6 +29,39 @@ int init_sim() {
...
+                return -1;
+            }
+        }
+    }
+
+    // Configure use of debugger
+    int pnum = 0;
+    char *s= getenv("HALIDE_HEX_DBG_PORT");
+    if (s && (pnum=(atoi(s)))) {
+        fprintf(stderr, "\nDebugger port: %d\n", pnum);
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/sim_host.cpp]: Also change this fprintf stderr to printf, and remove extra newline

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 5]
```diff
@@ -29,6 +29,39 @@ int init_sim() {
...
+    }
+
+    // Configure use of debugger
+    int pnum = 0;
+    char *s= getenv("HALIDE_HEX_DBG_PORT");
+    if (s && (pnum=(atoi(s)))) {
+        printf("Debugger port: %d\n", pnum);
+        HEXAPI_Status dbg_state;
+        dbg_state = sim->ConfigureRemoteDebug(pnum);
+        if (dbg_state) {
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/sim_host.cpp]: Should this be != HEX_STAT_SUCCESS?

Comment (ronlieb) [inline src/runtime/hexagon_remote/sim_host.cpp]: yes it should be.

**Pull request merged**